packages/browser/test/client.test.ts

```diff
@@ -1,38 +1,134 @@
-  const variants = testTimeoutNoRetriesClient.all();
+  const client = new ExperimentClient(API_KEY, {
+    retryFetchOnFailure: false,
+    fetchTimeoutMillis: 1,
+  });
+  await client.fetch(testUser);
+  const variants = client.all();
   expect(variants).toEqual({});
 });
 
-test('ExperimentClient.fetch, no retries, timeout failure', async () => {
-  await testTimeoutRetrySuccessClient.fetch(testUser);
-  const variant = testClient.variant('sdk-ci-test');
+test('ExperimentClient.fetch, with retries, retry success', async () => {
+  const client = new ExperimentClient(API_KEY, {
+    fetchTimeoutMillis: 1,
+  });
+  await client.fetch(testUser);
+  await delay(1000);
+  const variant = client.variant('sdk-ci-test');
+  expect(variant).toEqual({ value: 'on', payload: 'payload' });
+});
+
+test('ExperimentClient.fetch, fallbacks returned in correct order', async () => {
+  const client = new ExperimentClient(API_KEY, {
+    fallbackVariant: fallbackVariant,
+    initialVariants: initialVariants,
+  });
+
+  const firstFallbck: Variant = { value: 'first' };
+
+  let variant = client.variant('asdf', firstFallbck);
+  expect(variant).toEqual(firstFallbck);
+  variant = client.variant('asdf', 'first');
+  expect(variant).toEqual(firstFallbck);
+
+  variant = client.variant(initialKey, firstFallbck);
+  expect(variant).toEqual(firstFallbck);
+
+  variant = client.variant(initialKey);
+  expect(variant).toEqual(initialVariant);
+
+  await client.fetch(testUser);
+
+  variant = client.variant('asdf', firstFallbck);
+  expect(variant).toEqual(firstFallbck);
+  variant = client.variant('asdf', 'first');
+  expect(variant).toEqual(firstFallbck);
+
+  variant = client.variant(initialKey, firstFallbck);
+  expect(variant).toEqual(firstFallbck);
+
+  variant = client.variant(initialKey);
+  expect(variant).toEqual(initialVariant);
+
+  variant = client.variant(key, firstFallbck);
+  expect(variant).toEqual(serverVariant);
+});
+
+test('ExperimentClient.all, initial variants returned', async () => {
+  const client = new ExperimentClient(API_KEY, {
+    initialVariants: initialVariants,
+  });
+  const variants = client.all();
+  expect(variants).toEqual(initialVariants);
+});
+
+test('ExperimentClient.fetch, initial variants source, fetch overridden', async () => {
+  const client = new ExperimentClient(API_KEY, {
+    source: Source.InitialVariants,
+    initialVariants: initialVariants,
+  });
+  let variant = client.variant(key);
+  expect(variant).toEqual(offVariant);
+  await client.fetch(testUser);
+  variant = client.variant(key);
+  expect(variant).toEqual(offVariant);
+});
+
+test('ExperimentClient.fetch, sets user, setUser overrides', async () => {
+  const client = new ExperimentClient(API_KEY, {});
+  await client.fetch(testUser);
+  expect(client.getUser()).toEqual(testUser);
+  const newUser = { user_id: 'new_test_user' };
+  client.setUser(newUser);
+  expect(client.getUser()).toEqual(newUser);
```

Comment:
This part seems straight forward. How does doing a fetch before this affect things? Is there a side effect we should also be testing for?

Reply:
`fetch` with a user argument also sets the user within the SDK. This is probably the "feature" I like the least when it comes to using the SDK. 

So this test checks that `fetch` sets the user and that calling `setUser` after the fact overwrites whatever was set in `fetch`